lab-adam/server.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+const http = require('http');
+const url = require('url');
+const cowsay = require('cowsay');
+const parseBody = require('./lib/parse-body.js');
+const queryString = require('querystring');
+const PORT = process.env.PORT || 3000;
+
+http.createServer(function(req, res) {
+  req.url = url.parse(req.url);
+  req.url.query = queryString.parse(req.url.query);
+  console.log(req.url.query.text);
+  console.log(req.method);
+
+  if (req.url === '/'){
+    res.writeHead('200', {'Content-Type': 'text/plain'});
+    res.write('hello world');
+    res.end();
+  }
+
+  if (req.url.pathname === '/cowsay') {
+    switch(req.method){
+    case 'POST':
+      parseBody(req, res, function(err, data){
```

Comment:
You didn't check to see if the req.body is actually what you expect it to be (an object with a text property and a string value), only that it's valid JSON. You should have a condition that checks this, and if not (per the lab instructions), send back a 400 status response.